go/vt/vitessdriver/time.go

```diff
@@ -109,5 +109,16 @@ func NewDatetime(t time.Time, defaultLoc *time.Location) sqltypes.Value {
 	if t.Location() != defaultLoc {
 		t = t.In(defaultLoc)
 	}
-	return sqltypes.MakeTrusted(sqltypes.Datetime, []byte(t.Format(isoTimeFormat)))
+
+	// using this backing array and AppendFormat reduces heap allocations
+	var a [64]byte
+	var b = a[:0]
```

Comment:
Good catch. Since this variable is escaping, it's better to use `b := make([]byte, 0, 64)`

Reply:
I can't take any credit other than that I read articles by smart people. :) I just pushed a commit with your suggested change.